scripts/serverless.js

```diff
@@ -268,9 +268,25 @@ const processSpanPromise = (async () => {
             // If command was not recognized in previous resolution phases
             // parse args again also against schemas commands which require AWS service context
             resolveInput.clear();
+
+            // We are checking for `v` option being used before `resolveInput` here, because otherwise we
+            // lose context as `v` will be converted to `verbose` on returned `options` object
+            const usesVerboseAliasInOptions = Boolean(options.v);
```

Comment:
Is this the right place for this logic?

I think it's only triggered for AWS services, and only in certain scenarios

Reply:
According to our schema definition the commands that support `verbose` switch are only available for AWS services and only for `deploy`, `info`, `rollback` and `remove` command. I was checking the logic and testing it out for different scenarios and I believe it was the most reliable place to detect it, at previous resolution steps it wasn't resolved and there was no `commandSchema` matching available yet. 

Am I missing something here?

Reply:
At this point `providerName` may not be resolved yet (corner case of `provider` being behind not yet resolved vars).

e.g. there are two more steps where `providerName` resolution may finally occur:

https://github.com/serverless/serverless/blob/83af78ceb5b71885d2d2f349240e9154dac0f09d/scripts/serverless.js#L318-L321
https://github.com/serverless/serverless/blob/83af78ceb5b71885d2d2f349240e9154dac0f09d/scripts/serverless.js#L391

And there's also one extra step which may happen due local fallback:

https://github.com/serverless/serverless/blob/master/lib/Serverless.js#L282-L290

I assume best logic point to put it is here: 
https://github.com/serverless/serverless/blob/83af78ceb5b71885d2d2f349240e9154dac0f09d/lib/Serverless.js#L407

It'll also ensure that this deprecation is not shown with e.g. help command.

I totally understand it's confusing. It'll be way more cleaner in v3, when we will not have to dance so much with various deprecations

Reply:
Thanks for catching that, I didn't consider situation where provider is hidden behind variables. I've moved the logic into `Serverless` class instead as suggested.